frameless-runtime/src/amoeba.rs

```diff
@@ -183,3 +183,307 @@ impl Verifier for AmoebaCreation {
         Ok(0)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use tuxedo_core::dynamic_typing::testing::Bogus;
+
+    #[test]
+    fn creation_valid_transaction_works() {
+        let to_spawn = AmoebaDetails {
+            generation: 0,
+            four_bytes: *b"test",
+        };
+        let input_data = Vec::new();
+        let output_data = vec![to_spawn.into()];
+
+        assert_eq!(
+            AmoebaCreation.verify(&input_data, &output_data),
+            Ok(0),
+        );
+    }
+
+    #[test]
+    fn creation_invalid_generation_fails() {
+        let to_spawn = AmoebaDetails {
+            generation: 100,
+            four_bytes: *b"test",
+        };
+        let input_data = Vec::new();
+        let output_data = vec![to_spawn.into()];
+
+        assert_eq!(
+            AmoebaCreation.verify(&input_data, &output_data),
+            Err(VerifierError::WrongGeneration),
+        );
+    }
+
+    #[test]
+    fn creation_with_inputs_fails() {
+        let example = AmoebaDetails {
+            generation: 0,
+            four_bytes: *b"test",
+        };
+        let input_data = vec![example.clone().into()];
+        let output_data = vec![example.into()];
+
+        assert_eq!(
+            AmoebaCreation.verify(&input_data, &output_data),
+            Err(VerifierError::WrongNumberInputs),
+        );
+    }
+
+    #[test]
+    fn creation_with_badly_typed_output_fails() {
+        let input_data = Vec::new();
+        let output_data = vec![Bogus.into()];
+
+        assert_eq!(
+            AmoebaCreation.verify(&input_data, &output_data),
+            Err(VerifierError::BadlyTypedOutput),
+        );
+    }
+
+    #[test]
+    fn creation_multiple_fails() {
+        let to_spawn = AmoebaDetails {
+            generation: 0,
+            four_bytes: *b"test",
+        };
+        let input_data = Vec::new();
+        let output_data = vec![to_spawn.clone().into(), to_spawn.into()];
+
+        assert_eq!(
+            AmoebaCreation.verify(&input_data, &output_data),
+            Err(VerifierError::WrongNumberOutputs),
```

Comment:
Is it possible to make this be a little more granular or explain why it is a wrong number of outputs. This may be a nit and if you understand the amoeba in detail you should know why it is the wrong number of outputs. I am thinking of something like "TooManyChildren" or something like this. If it doesnt work in this context than can just disregard.